Kudu.Core/Deployment/DeploymentManager.cs

```diff
@@ -693,12 +693,18 @@ private DeployResult GetResult(string id, string activeDeploymentId, bool isDepl
                     {
                         await builder.Build(context);
                         builder.PostBuild(context);
-
-                        await RestartMainSiteIfNeeded(tracer, logger);
-
+                        if(deploymentInfo.Deployer == Constants.OneDeploy)
```

Comment:
Let's move the RestartMainSiteAsync() call inside RestartMainSiteIfNeeded() so that there is a clean abstraction.